copy/copy.go

```diff
@@ -361,6 +365,44 @@ func supportsMultipleImages(dest types.ImageDestination) bool {
 	return false
 }
 
+// compareImageDestinationManifestEqual compares the `src` and `dest` image's manifests, returning the 'true' if they compare equal.
+// The destination manifest is always fetched.
+func compareImageDestinationManifestEqual(ctx context.Context, options *Options, src types.Image, dest types.ImageDestination) (bool, []byte, string, digest.Digest, error) {
+
+	srcManifest, _, err := src.Manifest(ctx)
+	if err != nil {
+		return false, nil, "", "", errors.Wrapf(err, "Error reading manifest from image")
+	}
+
+	srcManifestDigest, err := manifest.Digest(srcManifest)
+	if err != nil {
+		return false, nil, "", "", errors.Wrapf(err, "Error calculating manifest digest")
+	}
+
+	destImageSource, err := dest.Reference().NewImageSource(ctx, options.DestinationCtx)
+	if err != nil {
+		return false, nil, "", "", nil
+	}
+
+	destManifest, destManifestType, err := destImageSource.GetManifest(ctx, nil)
+	if err != nil {
+		return false, nil, "", "", nil
+	}
+
+	destManifestDigest, err := manifest.Digest(destManifest)
+	if err != nil {
+		return false, nil, "", "", errors.Wrapf(err, "Error calculating manifest digest")
+	}
+
+	logrus.Debugf("Comparing source and destintion manifest digests: srcManifestDigest=%v, destManifestDigest=%v", srcManifestDigest, destManifestDigest)
+	if srcManifestDigest != destManifestDigest {
+		return false, nil, "", "", nil
+	}
+
+	// Destination and source manifests, types and digests should all be equivalent
+	return true, destManifest, destManifestType, srcManifestDigest, nil
```

Comment:
(Non-blocking: Using `destManifestDigest` here is exactly equivalent, but maybe a tiny bit easier to follow.)